src/lib.rs

```diff
@@ -355,8 +388,24 @@ pub fn run_tests<D: 'static + Send + Sync>(
     let mut num_benches = 0;
     let mut num_passed = 0;
 
+    let run_test = std::sync::Arc::new(run_test);
+    let run_test = move |test: &Test<D>| {
+        let run_test = run_test.clone();
+        std::panic::catch_unwind(std::panic::AssertUnwindSafe(move || {
+            run_test(test)
+        }))
+        .unwrap_or_else(|_e| {
+            Outcome::Failed { msg: None }
```

Comment:
Since capturing output is hard (see dedicated issue), we should at least already print the panic message.

```suggestion
        .unwrap_or_else(|e| {
            let msg = e.downcast_ref::<String>()
                .map(|s| s.as_str())
                .or(e.downcast_ref::<&str>().map(|s| *s));
            let error_msg = format!("test panicked: {}", msg);
            Outcome::Failed { msg: Some(error_msg) }
```